src/middlewared/middlewared/etc_files/loader.py

```diff
@@ -125,5 +126,19 @@ def generate_ec2_config(middleware):
         ]
 
 
+def generate_dual_nvdimm_config(middleware):
+    data = middleware.call_sync('system.dmidecode_info')['system-product-name']
+
+    product = data['system-product-name']
+    version = data['system-version']
+    minimum_vers = '3'
+
+    if product.startswith('TRUENAS-M') and version >= minimum_vers:
```

Comment:
Using string comparisons here is a bit odd, no?

Reply:
Casting it to float could lead to more issues if we start stamping with '3.0.1a' (not saying we will but who knows)

Local testing showed that python string comparison handled it correctly in about every combination that I could think of.

Reply:
system version is something we write, no?

Also there is `from packaging import version`

version.parse("3.0.1a")

Reply:
Neat little module that I didn't know existed. I've updated the PR.